Lib/annotationlib.py

```diff
@@ -183,12 +185,12 @@ def evaluate(
         else:
             code = self.__forward_code__
             try:
-                return eval(code, globals=globals, locals=locals)
+                return eval(code, globals=globals, locals={**type_param_scope, **locals})
             except Exception:
                 if not is_forwardref_format:
                     raise
             new_locals = _StringifierDict(
-                {**builtins.__dict__, **locals},
+                {**type_param_scope, **builtins.__dict__, **locals},
```

Comment:
This ordering is wrong, type params should override builtins.
```suggestion
                {**builtins.__dict__, **type_param_scope, **locals},
```
(Probably worth adding a test for this.)

Reply:
Yes, certainly. Logically I think it should go with the other tests for this particular code path (but they are in #138430), but happy to just add one in a new function if you'd prefer.